### PR TITLE
Rename EmptyTypTree into SyntacticEmptyTypeTree

### PR DESCRIPTION
**Pull request by densh**
1. Change the name as @xeno-by believes that previous name was misleading
2. Remove EmptyTree case as it's not needed any longer, only TypeTrees are used now 

review @xeno-by 
/cc @vladimirnik

### CODE REVIEW

**Comment by densh** [comment]
I wonder if we need would also want `tq""` to only match only trees without original. Currently they are just `TypeTree()` patterns without any restrictions.

**Comment by xeno-by** [comment]
@densh Absolutely.

File: src/reflect/scala/reflect/internal/BuildUtils.scala [hunk 1]
```diff
@@ -537,13 +537,9 @@ trait BuildUtils { self: SymbolTable =>
...
-    // abstract over possible alternative representations of no type in valdef
-    protected object EmptyTypTree {
-      def unapply(tree: Tree): Boolean = tree match {
-        case EmptyTree => true
-        case tt: TypeTree if (tt.original == null || tt.original.isEmpty) => true
-        case _ => false
-      }
+    // a type tree which wasn't specified by the user but might have been inferred by compiler
+    object SyntacticEmptyTypeTree {
+      def unapply(tt: TypeTree): Boolean = tt.original == null || tt.original.isEmpty
```
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/BuildUtils.scala:548]: What's the difference with `tt.wasEmpty`? /cc @retronym

Comment (densh) [inline src/reflect/scala/reflect/internal/BuildUtils.scala:548]: `tt.wasEmpty` isn't set after parsing and thus can't be used here. @retronym is it meant to be this way?

**Comment by xeno-by** [comment]
@densh Do you plan to submit the changes to how `tq""` works as part of this pull request?

**Comment by densh** [comment]
@xeno-by yes, will submit them soon today.

**Comment by xeno-by** [comment]
LGTM

**Pull request merged**